src/Symfony/Component/DomCrawler/FormFieldRegistry.php

```diff
@@ -124,13 +124,15 @@ public function has($name)
     public function set($name, $value)
     {
         $target = &$this->get($name);
-        if (!is_array($value) || $target instanceof Field\ChoiceFormField) {
+        if ($target instanceof Field\FormField) {
```

Comment:
I don't think it's cast to string, it simply sets the array as a value. I'll add this test case. Cheers!